app/models/shipit/deploy.rb

```diff
@@ -282,8 +283,13 @@ def update_last_deploy_time
       stack.update(last_deployed_at: ended_at)
     end
 
+    def mark_deploy_ref_update
+      @update_ref = true
+    end
+
     def update_latest_deployed_ref
-      stack.update_latest_deployed_ref
+      stack.update_latest_deployed_ref if @update_ref
+      @update_ref = false
```

Comment:
Rather than use a boolean flag you can check the dirty attributes: https://github.com/Shopify/shipit-engine/blob/626845cc540bffbaf122f7eb3e6044bbd069ed15/app/models/shipit/stack.rb#L549-L553

Reply:
Oh, that's awesome. Thanks!